ext/pcntl/pcntl.c

```diff
@@ -1012,6 +1017,29 @@ PHP_FUNCTION(pcntl_signal)
 }
 /* }}} */
 
+/* {{{ proto bool pcntl_signal_get_handler(int signo)
+   Gets signal handler */
+PHP_FUNCTION(pcntl_signal_get_handler)
+{
+	zval *prev_handle;
+	zend_long signo;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &signo) == FAILURE) {
+		return;
+	}
+
+	if (signo < 1 || signo > 32) {
```

Comment:
I was copy-pasting that via pcntl_signal().  The problem with using NSIG is that on my box(linux 4.4.0, libc 2.22-4) it's value is 64, however 32 is the largest defined signal.  That said, NSIG on FreeBSD is defined to 32, even though it has SIGLIBRT defined at 33.  So, I'm not sure what happens if you're to sigaction() a signal that's not defined in signal.h, or if in general we might look to be smarter about what is the highest signal we support per-platform.

Regardless, this could be something addressed in a distinct PR trying to abstract and remove the hard-coded values.